Tools/ViewsEstimators.py

```diff
@@ -6,118 +6,333 @@
+
+                    'GBMRegressor': GradientBoostingRegressor(n_estimators=200, random_state=random_state),
+                    'GBMClassifier': GradientBoostingClassifier(n_estimators=200, random_state=random_state),
+
+                    'XGBRegressor': XGBRegressor(n_estimators=100, learning_rate=0.05, random_state=random_state),
+                    'XGBClassifier': XGBClassifier(n_estimators=100, learning_rate=0.05, random_state=random_state),
+                    
+                    'HGBRegressor': HistGradientBoostingRegressor(max_iter=200, random_state=random_state),
+                    'HGBClassifier': HistGradientBoostingClassifier(max_iter=200, random_state=random_state),
                 }
+        estimator = estimators.get(estimator_name)
 
-        return funcs[func_name]
+        return estimator
 
+    # Define the fit method for the class to train the model
     def fit(self,
             X: Union[np.ndarray, pd.DataFrame],
             y: Union[np.ndarray, pd.Series]):
-        X, y = check_X_y(X, y, dtype=None,
+
+        # Use sklearn fucntionality to check X and y for consistent length and enforce X to be 2D and y 1D. 
+        # By default, X is checked to be non-empty and containing only finite values.
+        # Standard input checks are also applied to y, such as checking that y
+        # does not have np.nan or np.inf targets.
+        X, y = check_X_y(X, 
+                         y, 
+                         dtype=None,
                          accept_sparse=False,
                          accept_large_sparse=False,
-                         force_all_finite='allow-nan')
+                         force_all_finite=True) #'allow-nan'
+        
+        # Set the number of features seen during fit
+        self.n_features_in_ = X.shape[1]
 
         if X.shape[1] < 2:
             raise ValueError('Cannot fit model when n_features = 1')
 
-        self.clf_ = self._resolve_estimator(self.clf_name)
+        # Instantiate the classifier
+        self.clf_ = self._resolve_estimator(self.clf_name, random_state=42)
         if self.clf_params:
             self.clf_.set_params(**self.clf_params)
-        self.clf_.fit(X, y > 0)
-        self.clf_fi = self.clf_.feature_importances_
+        
+        # Check if there are more than one unique values in y 
+        # and if yes, fit the classifier to X, so y that is > 0, becomes 1
+        # and if not, it is 0
+        if len(np.unique(y)) > 1:
+            self.clf_.fit(X, y > 0)
+        else:
+            # Handle the case where y has only one unique value
+            self.clf_ = DummyClassifier(strategy='most_frequent')
+            self.clf_.fit(X, y > 0)
+
 
-        self.reg_ = self._resolve_estimator(self.reg_name)
+        # Instantiate the regressor
+        self.reg_ = self._resolve_estimator(self.reg_name, random_state=42)
         if self.reg_params:
             self.reg_.set_params(**self.reg_params)
```

Comment:
THIS IS HOW FAR I'VE COME NOW. I'LL REVIEW THE REST LATER (capital for attention)